test/functional/treesitter/parser_spec.lua

```diff
@@ -506,6 +506,34 @@ int x = INT_MAX;
       end)
     end)
 
+    describe("when providing parsing information through a directive", function()
+      it("should inject a language", function()
+        exec_lua([=[
+        vim.treesitter.add_directive("inject-clang!", function(match, _, _, pred, metadata)
```

Comment:
It'd be good to add some more information / documentation about this I think.

Reply:
I can add some info to the vim docs

Reply:
Done!